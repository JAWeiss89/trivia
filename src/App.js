import React, {useState} from 'react';
import TriviaGame from './TriviaGame';
import WelcomeScreen from './WelcomeScreen';
import './styles/App.css';

function App() {
  let [beginGame, setBeginGame] = useState(false);
  let [name, setName] = useState(null);

  return (
    <div className="App">
      {beginGame 
      ? <>
        <h2><span className="tandem">tandem</span>trivia</h2>
        <TriviaGame name={name}/>
        </>
      : <WelcomeScreen setName={setName} setBeginGame={setBeginGame}/>
      }
      <p className="credits">Designed & Developed by <a href="https://www.JorgeWeiss.com">Jorge Weiss</a></p>
    </div>
  );
}

export default App;
