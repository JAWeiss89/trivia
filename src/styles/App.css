.App * {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  text-align: center;
  background-color: rgb(232 232 232);
  color: rgb(51, 51, 51);
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.App a {
  text-decoration: none;
  color: inherit;
}

.App h2 {
  font-family: 'Roboto Mono', monospace;
  position: absolute;
  top: 10px;
  right: 10px;
  font-weight: 400;
}


button {
  height: 50px;
  width: 150px;
  display: block;
  font-size: 24px;
  border: none;
  border-radius: 10px;
  background-color: rgb(172, 172, 172);
  color: rgb(65, 65, 65);
}

