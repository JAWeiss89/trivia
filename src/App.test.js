import { render, fireEvent } from '@testing-library/react';
import App from './App';

// smoke test
test('renders learn react link', () => {
  render(<App />);
});

// snapshot test
test("matches snapshot", () => {
  const { asFragment } = render( <App />);
  expect(asFragment()).toMatchSnapshot();
});

// other tests
test("correct elements are present on load", () => {
  const {queryByText, queryByPlaceholderText} = render(<App />)

  //expect WelcomeScreen to show on initial render
  expect(queryByText("trivia")).toBeInTheDocument();
  expect(queryByPlaceholderText("name")).toBeInTheDocument();

})
test("submits form and goes to game screen", () => {
  const {queryByText, queryByPlaceholderText} = render(<App />)


  //fill name field and submit
  const nameField = queryByPlaceholderText("name");
  const startBtn = queryByText("start");
  fireEvent.change(nameField, { target: { value: 'jorge' } })
  fireEvent.click(startBtn);

  expect(queryByText("Score: 0")).toBeInTheDocument();
})