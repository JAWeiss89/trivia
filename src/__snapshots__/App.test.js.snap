// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="WelcomeScreen"
    >
      <h1>
        <span
          class="tandem"
        >
          tandem
        </span>
        trivia
      </h1>
      <form
        autocomplete="off"
      >
        <input
          name="name"
          placeholder="name"
          type="text"
          value=""
        />
        <button
          class="none"
        >
          start
        </button>
      </form>
    </div>
    <p
      class="credits"
    >
      Designed & Developed by 
      <a
        href="https://www.JorgeWeiss.com"
      >
        Jorge Weiss
      </a>
    </p>
  </div>
</DocumentFragment>
`;
